graphql_compiler/compiler/ir_lowering_common/common.py

```diff
@@ -249,7 +268,9 @@ def extract_folds_from_ir_blocks(ir_blocks):
     return folds, remaining_ir_blocks
 
 
-def extract_optional_location_root_info(ir_blocks):
+def extract_optional_location_root_info(
+    ir_blocks: List[BasicBlock],
+) -> Tuple[List[BaseLocation], Dict[BaseLocation, Tuple[BaseLocation, ...]]]:
```

Comment:
Selene raised a good point in my PR [here](https://github.com/kensho-technologies/graphql-compiler/pull/834#discussion_r436807755) that when type-hinting functions, we can also revise the Args and Returns descriptions in the docstring to describe only what the args and return values do (if it's not particularly inconvenient), rather than explicitly rewriting their type in the docstring.

Reply:
Makes sense. I think I'm going to avoid updating the Args/Returns sections and only focus on adding type hints in this PR, though, especially since the signature is staying functionally the same.